File: app/src/main/java/theo/base/AppInitializer.java
package theo.base;


import com.squareup.leakcanary.LeakCanary;

import net.danlew.android.joda.JodaTimeAndroid;

public abstract class AppInitializer {

    void init(BaseApplication app){
        JodaTimeAndroid.init(app);
        LeakCanary.install(app);
        buildTypeDependentInitialization(app);
    }

    abstract void buildTypeDependentInitialization(BaseApplication app);
}

File: app/src/main/kotlin/theo/base/AppInitializer.kt
package theo.base


import com.squareup.leakcanary.LeakCanary

import net.danlew.android.joda.JodaTimeAndroid

abstract class AppInitializer {

    fun init(app: BaseApplication) {
        JodaTimeAndroid.init(app)
        LeakCanary.install(app)
        buildTypeDependentInitialization(app)
    }

    protected abstract fun buildTypeDependentInitialization(app: BaseApplication)
}
